clients/python-wrapper/lakefs/config.py

```diff
@@ -48,8 +61,42 @@ class Credentials(LenientNamedTuple):
         access_key_id: str
         secret_access_key: str
 
-    server: Server
-    credentials: Credentials
+    class AuthType(Enum):
+        """
+        Enum for the supported authentication types
+        """
+        SESSION_TOKEN = 1
+        CREDENTIALS = 2
+        IAM = 3
+
+    class ProviderType(Enum):
+        """
+        Enum for the supported authentication provider types
+        """
+        AWS_IAM = "aws_iam"
+        UNKNOWN = "unknown"
+
+    class AWSIAMProviderConfig(LenientNamedTuple):
```

Comment:
Any reason why this is a `LenientNamedTuple`?
This class is used to be able to initialize structs using `**kwargs` without having to worry about it failing due to unknown keys